src/main/java/com/limechain/network/Network.java

```diff
@@ -141,14 +139,31 @@ public String[] getListenAddresses() {
         // TODO Bug: .listenAddresses() returns empty list
         return this.host.listenAddresses().stream().map(Multiaddr::toString).toArray(String[]::new);
     }
+    public int getPeersCount() {
+        return kademliaService.getAddresses().size();
+    }
 
     public List<PeerAddresses> getPeers() {
-        List<PeerAddresses> peers = new ArrayList<>();
-        for (Map.Entry<Multihash, List<MultiAddress>> entry : connections.entrySet()) {
-            PeerAddresses peer = new PeerAddresses(entry.getKey(), entry.getValue());
-            peers.add(peer);
-        }
-        return peers;
+        return kademliaService.getAddresses()
+                .entrySet()
+                .stream()
+                .map(this::mapAddressBookEntryToPeerAddresses)
+                .toList();
+    }
```

Comment:
Since `getPeers` is not used anywhere currently, I'd suggest to remove this functionality. (it's going to be used for the `system_peers` RPC endpoint in the near future)